File: Hashing/Cpp/Frequency.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int N;
    cin>>N;
    int arr[N];
    unordered_map<int, int> hash_map;
    for(int i=0; i<N; i++){
        cin>>arr[i];
        hash_map[arr[i]] = hash_map[arr[i]] + 1;
    }
    cout<<"Frequencies are : "<<endl;
    for(auto e : hash_map){
        cout<< e.first <<" : "<< e.second <<endl;
    }
}

File: Hashing/Frequency.java
import java.util.*;

class FrequencyOccurence{
  public static void main(String[] args){
      Scanner sc = new Scanner(System.in);
      int N = sc.nextInt();
      int arr[] = new int[N];
      Map<Integer, Integer> map = new HashMap<>();
      for(int i=0; i<N; i++){
          arr[i] = sc.nextInt();

          if(map.containsKey(arr[i])){
              map.put(arr[i], map.get(arr[i]) + 1);
          }
          else{
              map.put(arr[i], 1);
          }
      }
      System.out.println("Frequencies are : ");
      for(int e : map.keySet()){
          System.out.println(e +" : "+ map.get( e));
      }      
  }
}
